### PR TITLE
[server] AAWC should ignore NO_OP new field from update record

### PR DESCRIPTION
**Pull request by sixpluszero**
## [server] AAWC should ignore NO_OP new field from update record
If a field from update record is new field but it is NO_OP, we should skip this field, instead of letting it slip into checking TS from old TS record, which will result in exception and halt the ingestion

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java [hunk 1]
```diff
@@ -766,9 +766,11 @@ private boolean ignoreNewUpdate(
       case PER_FIELD_TIMESTAMP:
         GenericRecord timestampRecord = (GenericRecord) oldTimestampObject;
```
Comment (majisourav99) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java:767]: unrelated nit: can we rename this as `oldTimestampRecord` ?

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java [hunk 2]
```diff
@@ -766,9 +766,11 @@ private boolean ignoreNewUpdate(
...
         for (Schema.Field field: writeComputeRecord.getSchema().getFields()) {
-          if (getFieldOperationType(writeComputeRecord.get(field.pos())) != NO_OP_ON_FIELD
-              && timestampRecord.get(field.name()) == null) {
-            return false; // Write Compute tries to update a non-existing field.
+          if (!timestampRecord.hasField(field.name())) {
+            if (getFieldOperationType(writeComputeRecord.get(field.pos())) == NO_OP_ON_FIELD) {
+              continue; // New field does not perform actual update.
+            }
+            return false; // Partial update tries to update a non-existing field.
```
Comment (majisourav99) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java:773]: should this be true ie be ignored ?

Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java:773]: True means we should ignore the whole update.
continue means we ignore this field, but whether to ignore the whole UPDATE depends on the check of other fields.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java (hunk 1 above)
Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java:767]: Sure, I can update.

**Review by majisourav99 — APPROVED** [review]
Thanks a lot Jialin for the quick fix

**Pull request merged**